ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -1030,6 +1037,315 @@ private Pair<ServerErrorCode, GetResponse> getBlobFromRemoteReplica(ReplicateBlo
     }
   }
 
+  // LOCAL_CONSISTENCY_TODO handleReplicateBlobRequestV2 will replace handleReplicateBlobRequest totally when it's verified.
+  public void handleReplicateBlobRequestV2(NetworkRequest request) throws IOException, InterruptedException {
+    if (connectionPool == null || transformer == null || transformer.get() == null) {
+      throw new UnsupportedOperationException("ReplicateBlobRequest is not supported on this node.");
+    }
+
+    ReplicateBlobRequest replicateBlobRequest;
+    if (request instanceof LocalChannelRequest) {
+      // This is a case where handleReplicateBlobRequest is called when frontends are talking to Azure. In this case, this method
+      // is called by request handler threads running within the frontend router itself. So, the request can be directly
+      // referenced as java objects without any need for deserialization.
+      replicateBlobRequest = (ReplicateBlobRequest) ((LocalChannelRequest) request).getRequestInfo().getRequest();
+    } else {
+      replicateBlobRequest = ReplicateBlobRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    }
+    long startProcessTime = SystemTime.getInstance().milliseconds();
+    // Get the parameters from the replicateBlobRequest: the blobId, the remoteHostName and the remoteHostPort.
+    BlobId blobId = replicateBlobRequest.getBlobId();
+    String remoteHostName = replicateBlobRequest.getSourceHostName();
+    int remoteHostPort = replicateBlobRequest.getSourceHostPort();
+
+    // check the request parameters.
+    if (replicateBlobRequest.getOperationType() != RequestOrResponseType.PutRequest
```

Comment:
We are not going to repair a PUT request with replicate blob request. I think we should remove the PutRequest here.

Reply:
ODR V1 version is replicating the PutBlob. By default the operationType is PutBlob. For example, if Ttlupdate partially  failed and some replica returned NOT_FOUND, we replicate the PutBlob from the source replica and then we retry the frontend TtlUpdate request.
The code supports both cases, the current ODR and future local blob consistency case. Without frontend code change, this code still works with ODR V1.

Reply:
The planned deployment order will be,
1. Deploy this server code.
2. Frontend modification to utilize V2 and deploy to the frontend cluster.
3. Server V1 related code cleanup if needed.

Reply:
Added the comment.